src/source/tile.js

```diff
@@ -73,6 +76,7 @@ class Tile {
     texture: any;
     refreshedUponExpiration: boolean;
     reloadCallback: any;
+    crossTileSymbolIndex: any;
```

Comment:
Do tiles need to hold a reference to the `CrossTileSymbolIndex`? I added this because it was the easiest way to get access to it in the added/removed callbacks but it might be good to get rid of this if possible.

Reply:
`SourceCache` is already holding a reference to the map (and thus the painter and the `CrossTileSymbolIndex`), so we could pass this in as an argument to `added` and `removed`. I'll do that.

Reply:
Done.